src/test/java/org/monarchinitiative/fhir2hpo/codesystems/CodeableConceptAnalyzerTest.java

```diff
@@ -33,29 +33,38 @@ public void testOneDistinctCode() throws UnmappedCodeableConceptException, Confl
 		assertEquals("Expected internal code 'L' for external codes 'LL' and 'L'", "L", internalCode.name());
 	}
 
-	@Test(expected = UnmappedCodeableConceptException.class)
-	public void testUnmappedSystem() throws UnmappedCodeableConceptException, ConflictingInternalCodesException {
+	public void testUnmappedSystem() {
 		List<Coding> codings = new ArrayList<>();
 		codings.add(FhirMockUtils.mockCoding("Other System", "LL"));
 		CodeableConcept codeableConcept = FhirMockUtils.mockCodeableConcept(codings);
-		CodeableConceptAnalyzer.getInternalCodeForCodeableConcept(codeableConcept);
+		try {
+			CodeableConceptAnalyzer.getInternalCodeForCodeableConcept(codeableConcept);
+		} catch (ConversionException e) {
+			assertEquals("Expected the system to be unmapped", ConversionExceptionType.UNMAPPED_CODEABLE_CONCEPT, e.getType());
+		}
 	}
 
-	@Test(expected = UnmappedCodeableConceptException.class)
-	public void testUnmappedCode() throws UnmappedCodeableConceptException, ConflictingInternalCodesException {
+	public void testUnmappedCode() {
 		List<Coding> codings = new ArrayList<>();
 		codings.add(FhirMockUtils.mockCoding("http://hl7.org/fhir/v2/0078", "Other Code"));
 		CodeableConcept codeableConcept = FhirMockUtils.mockCodeableConcept(codings);
-		CodeableConceptAnalyzer.getInternalCodeForCodeableConcept(codeableConcept);
+		try {
+			CodeableConceptAnalyzer.getInternalCodeForCodeableConcept(codeableConcept);
+		} catch (ConversionException e) {
+			assertEquals("Expected the code to be unmapped", ConversionExceptionType.UNMAPPED_CODEABLE_CONCEPT, e.getType());
+		}
```

Comment:
You might want to consider using the `ExpectedException` rule, which allows making nice assertions about exceptions without using a try / catch:

https://github.com/junit-team/junit4/wiki/exception-testing#expectedexception-rule

Reply:
This would be useful if we go with the hierarchy of exceptions as well since we could more specifically the detect the exception.

Reply:
Yes, I looked at using the Rule, but it doesn't work with this design, since I'm unable to assert on the underlying enumeration returned with the exception. As @kirenotneb points out, if we change to a hierarchy I would be able to use that.